README.md

```diff
@@ -38,79 +32,409 @@ IncludeOS is free software, with "no warranties or restrictions of any kind".
+- Then start implementing the `Service::start` function in the `Service` class, located in `<your-service-name>/service.cpp` (very simple example provided). This function will be called once the OS is up and running.
+- Update the `CMakeLists.txt` to specify the name of your project, enable any needed drivers or plugins, etc.
+- Update the `conanfile.txt` to specify any build requirements and generators.
+
+You should then be able to run your service in the same way as the demo_service.
+
+To add a new service to our demo examples, make a PR to our [demo-examples](https://github.com/includeos/demo-examples) repo. Make sure to
+add a README in your example folder with description of your service.
+
+**Note:** Remember to deactivate your service environment after your work with:
+
+```
+  $ source deactivate.sh
+```
+
+<!-- TODO: ### How building with Conan works? -->
+___
+
+### Getting IncludeOS in editable mode
+
+To get started with getting the conan package in editable mode,
+
+- ###### Edit the `conanfile.py` : comment out the `version = get_version()`
+
+- ###### Create a `build` folder to build IncludeOS
+
+- ###### Edit the `layout.txt` : _(uses jinja2 syntax)_ needed to parse the build folder correctly.
+
+Do some local adaptions for where your build folder is by setting your build
+folder in the first line as follows:
+
+```
+  {% set build_dir='build' %}
+  [bindirs]
+  cmake
+  [includedirs]
+  api
+  [libdirs]
+  {{ build_dir }}/plugins
+  {{ build_dir }}/drivers
+  {{ build_dir }}/lib
+  [resdirs]
+  {{ build_dir }}
+
+```
+
+- ###### Set the conan package into **editable mode**
+
+Make sure to adjust the version to whatever is apropriate.
+
+```
+  conan editable add . includeos/0.15.0@includeos/test --layout=layout.txt
+```
+**Note:** Avoid choosing `latest`
+
+- ###### Check Status
+
+The package is now in **editable mode** and any dependencies of IncludeOS will
+pick this IncludeOS package from your local cache. If you make any changes to the
+code a simple `make` should be enough. However if your dependencies have changed
+you need to redo the `conan install` step.
+
+Here is an example on how it looks when its pulled into cache as editable:
+
+```
+  $ conan editable list
+  includeos/0.15.0@includeos/test:45955af12a7f7608830c1d255b80a75440406e3c - Editable
+```
+
+- ###### Build IncludeOS
+
+```
+  cd build
+  conan install .. -pr <conan_profile> (-o options like solo5=True etc)
+  cmake .. (-DTOOLCHAIN_PROFILE=xx.yy when needed)
+  make
+```
+
+- ###### Finalizing Changes
+
+Once the code is **finalized** and you want to verify that the conan package
+still builds remove the editable:
+
+```
+  $ conan editable remove includeos/0.15.0@includeos/test
+```
+
+Then remove the comment on the `#version` in the `conanfile.py` and do a normal
+
+```
+  $ conan create <source_path> includeos/test -pr <conan_profile>
+```
+
+___
+
+### Getting started developing packages
```

Comment:
I think it would be useful with more details around when you need this -- I guess the main use case is writing a service or driver in a separate repo that depends on includeos?

Reply:
The plan is to add more details to the service/driver repos and more in the docs. For now I have updated the text a bit.